packages/ui/react-ui-editor/src/components/Toolbar/Toolbar.tsx

```diff
@@ -89,9 +102,14 @@ const ToolbarButton = ({ Icon, onClick, title }: ToolbarButtonProps) => {
 const ToolbarSeparator = () => <div className='grow' />;
 
 const MarkdownHeading = () => {
-  const { onAction } = useToolbarContext('MarkdownFormatting');
+  const { onAction, state } = useToolbarContext('MarkdownFormatting');
+  const header = state.blockType && /heading(\d)/.exec(state.blockType);
+  const value = header ? header[1] : state.blockType === 'paragraph' || !state.blockType ? '0' : null;
+  if (value === null) {
+    return <></>;
```

Comment:
return null to render nothing. is that the intention though? shouldn't the select be always visible?

Reply:
If you're in a table cell or code block, this control can't really do a lot. I could disable it instead, if you prefer.

Reply:
yes i think better than the visible editor changing shape; for tootips it should be removed as you've done.